nav2_costmap_2d/src/layer.cpp

```diff
@@ -84,8 +84,8 @@ void
 Layer::undeclareAllParameters()
 {
   std::for_each(begin(local_params_), end(local_params_), [this](const std::string & param_name) {
```

Comment:
std::for_each makes no sense here. This should just be a simple for loop.

```suggestion
  for(auto & param_name : local_params_) {
    node_->undeclare_parameter(getFullName(param_name));
  }
```